cantools/database/can/conversion.py

```diff
@@ -0,0 +1,190 @@
+
+
+class IdentityConversion(BaseConversion):
+    scale = 1
+    offset = 0
+    choices = None
+
+    def __init__(self, is_float: bool) -> None:
+        self.is_float = is_float
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> Union[int, float]:
+        return raw_value
+
+    def scaled_to_raw(self, scaled_value: SignalValueType) -> Union[int, float]:
+        if not isinstance(scaled_value, (int, float)):
+            raise TypeError(
+                f"'scaled_value' must have type 'int' or 'float' (is {type(scaled_value)})"
+            )
+        return scaled_value
+
+
+class LinearIntegerConversion(BaseConversion):
+    is_float = False
+    choices = None
+
+    def __init__(self, scale: int, offset: int) -> None:
+        self.scale: int = scale
+        self.offset: int = offset
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> SignalValueType:
+        return raw_value * self.scale + self.offset
+
+    def scaled_to_raw(self, scaled_value: SignalValueType) -> Union[int, float]:
+        if not isinstance(scaled_value, (int, float)):
+            raise TypeError(
+                f"'scaled_value' must have type 'int' or 'float' (is {type(scaled_value)})"
+            )
+        return round((scaled_value - self.offset) / self.scale)
+
+
+class LinearConversion(BaseConversion):
+    choices = None
+
+    def __init__(self, scale: float, offset: float, is_float: bool) -> None:
+        self.scale = scale
+        self.offset = offset
+        self.is_float = is_float
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> SignalValueType:
+        return raw_value * self.scale + self.offset
+
+    def scaled_to_raw(self, scaled_value: SignalValueType) -> Union[int, float]:
+        if not isinstance(scaled_value, (int, float)):
+            raise TypeError(
+                f"'scaled_value' must have type 'int' or 'float' (is {type(scaled_value)})"
+            )
+        _raw = (scaled_value - self.offset) / self.scale
+        if self.is_float:
+            return _raw
+        return round(_raw)
+
+
+class NamedSignalConversion(BaseConversion):
+    is_float = False
+
+    def __init__(self, scale: float, offset: float, choices: Choices) -> None:
+        self.scale = scale
+        self.offset = offset
+        self._inverse_choices: Optional[Dict[str, int]] = None
+        self.choices = choices
+        self._update_choices()
+
+        self._conversion = conversion_factory(
+            scale=self.scale,
+            offset=self.offset,
+            choices=None,
+            is_float=False,
+        )
+
+    def raw_to_scaled(
+        self,
+        raw_value: Union[int, float],
+        decode_choices: bool = True,
+    ) -> SignalValueType:
+        if decode_choices:
```

Comment:
`self.choices` can be `None` here (?):

```suggestion
        if decode_choices and self.choices:
```

Reply:
Choices should be mandatory for NamedSignalConversion, so that check is unnecessary